org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/internal/core/breakpoints/JavaBreakpoint.java

```diff
@@ -85,15 +96,14 @@ public void addToVMTarget(IVMTarget target) {
         }
 
         if (referenceTypeName.indexOf("$") == -1) {
-            target.getEventHub().addJDIEventListener(createClassPrepareRequest(referenceTypeName, null, target), this);
+            this.registerRequest(createClassPrepareRequest(referenceTypeName, null, target), target);

```

Comment:
registerClassPrepareRequest is better to avoid duplicate createClassPrepareRequest statements.